crates/libs/bindgen/src/gen.rs

```diff
@@ -156,13 +179,14 @@ impl<'a> Gen<'a> {
             Type::WinrtArray(ty) => self.type_name(ty),
             Type::WinrtArrayRef(ty) => self.type_name(ty),
             Type::ConstRef(ty) => self.type_name(ty),
-            _ => unimplemented!(),
+            Type::TypeRef(TypeDefOrRef::TypeRef(ref_)) => self.type_ref_name(*ref_),
```

Comment:
Note that code gen from `TypeRef` is not a good idea as it implies the `TypeDef` is not available and thus you have no idea what kind of type you're actually generating code for. Usually this implies you're missing metadata. I should have better diagnostics for this in `riddle` soon.

Reply:
Throwing in my experiences from last night smoke testing riddle. I ran into this too; I was missing base Win32 Metadata winmd in the riddle input folder.

Reply:
@kennykerr ah, thanks, so you're basically saying that I fixed this in the wrong way. Which might match the [observation of `Foundation` having vanished](https://github.com/microsoft/windows-rs/pull/2539#issuecomment-1591977671).

Improved diagnostics (even something simple as a format-string in an `unimplemented!()` call) are definitely welcome!

@riverar note that this PR was from before `riddle` and might be caused by the Win32metadata updates instead.

Reply:
> Note that code gen from `TypeRef` is not a good idea as it implies the `TypeDef` is not available and thus you have no idea what kind of type you're actually generating code for. Usually this implies you're missing metadata. I should have better diagnostics for this in `riddle` soon.

@kennykerr are you still planning on improving the diagnostics here? A similar issue is appearing again because of incomplete metadata: https://github.com/microsoft/win32metadata/issues/1760

Reply:
Please open a new issue and include a repro. Thanks.

Reply:
Reported at #2742. The repro is generating new metadata with `win32metadata` `main` (relevant commit hash is linked in case this is fixed upstream), followed by running `tool_windows`.